api/server/server.go

```diff
@@ -127,15 +131,37 @@ func GetClusterAPIRoutes() []*Route {
 	return clusterApi.Routes()
 }
 
+func SetClusterAPIRoutesWithAuth(router *mux.Router, authenticators map[string]auth.Authenticator) (*mux.Router, error) {
+	clusterApi := newClusterAPI()
+	return clusterApi.SetupRoutesWithAuth(router, authenticators)
+}
+
+func GetCredentialsRoutes() []*Route {
+	return (&volAPI{}).credsRoutes()
+}
+
+func SetCredentialsRoutesWithAuth(router *mux.Router, authenticators map[string]auth.Authenticator) (*mux.Router, error) {
```

Comment:
Just curious why we have just the credentials here? Will we have other routes other than cluster here in this file also? You may want to add some comments for each of these to explain to the reader why these functions are here.

Reply:
Yes, I think I should make better name for this PR, it contains securing not only cluster routes.  I've made this method for porx actually. Porx does not only start Volume management server, but it also just takes routes from it to start in its own REST server.

Here:

https://github.com/portworx/porx/pull/4789/files#diff-a38fd22476d6f900fdd9f3354cdb429fR337

So I've made this helper method to cover cred routes when porx get them.

https://github.com/portworx/porx/pull/4789/files#diff-a38fd22476d6f900fdd9f3354cdb429fR150